core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt

```diff
@@ -1150,4 +1192,43 @@ class TestCaseWriter {
                 && !printableContent.contains("logged")
                 && !printableContent.contains("""\w+:\d{4,5}""".toRegex()))
     }
+
+    fun getPrintableInputGenes(printableInputGene: MutableList<String>): String {
+        var printableInputGenes = ""
+        for (elt in printableInputGene) {
+            printableInputGenes = "$elt,$printableInputGenes"
+        }
+        printableInputGenes = printableInputGenes.substring(0, printableInputGenes.length - 1)
+        printableInputGenes = printableInputGenes.replace("\"", "\\\"")
+        return printableInputGenes
+    }
+
+    fun getPrintableInputGene(inputGenes: List<Gene>): MutableList<String> {
+        val printableInputGene: MutableList<String> = mutableListOf()
+        for (gene in inputGenes) {
+            if (gene is EnumGene<*>) {
+                val i = gene.getValueAsRawString()
+                printableInputGene.add("${gene.name} : $i")
+            } else {
+                val i = gene.getValueAsPrintableString(mode = GeneUtils.EscapeMode.GQL_INPUT_MODE)
+                printableInputGene.add("${gene.name} : $i")
+            }
+        }
+        return printableInputGene
+    }
+
+    fun getMutation(returnGene: Gene, a: GraphQLAction): String {
+        var mutation = "{${returnGene.getValueAsPrintableString(mode = GeneUtils.EscapeMode.BOOLEAN_SELECTION_MODE)}}".replace("{${a.methodName}", "")
+        mutation = mutation.substring(0, mutation.length - 1)
+        return mutation
+    }
+
+    fun getQuery(returnGene: Gene, a: GraphQLAction): String {
+        var query = "{${returnGene.getValueAsPrintableString(mode = GeneUtils.EscapeMode.BOOLEAN_SELECTION_MODE)}}"
+        query = query.replace("{${a.methodName}", "", true)
+        query = query.substring(0, query.length - 1)
+        return query
```

Comment:
no need to re-assign query:
```
var query = "{${returnGene.getValueAsPrintableString(mode = GeneUtils.EscapeMode.BOOLEAN_SELECTION_MODE)}}"
        .replace("{${a.methodName}", "", true)
        .substring(0, query.length - 1)
```

Reply:
i m bit concerned about this `replace` calls... what if the same value is repeated inside a string? eg, when `a.methodName` is a generic term. furthermore, should add some comments to specify why this is done

Reply:
I agree. I did not find another solution. I tried to remove the name (return param name) from the action builder it throws the exception: empty name.
Comments->done

Reply:
why not handling it inside the `getValueAsPrintableString`? could either be based on `mode`, or can add new param to that function

Reply:
1.why not handling it inside the getValueAsPrintableString? -> if I put it inside: getValueAsPrintableString I will lost all the objects names inside the return object gene since it is printed recursively.
2.could either be based on mode, or can add new param to that function->I did not understand those two propositions